lib/phoenix/transports/websocket.ex

```diff
@@ -32,6 +32,7 @@ defmodule Phoenix.Transports.WebSocket do
         case handler.connect(config) do
           {:ok, state} -> {:ok, conn, state}
           :error -> {:error, Plug.Conn.send_resp(conn, 403, "")}
+          {:error, status, body} -> {:error, Plug.Conn.send_resp(conn, status, body)}
```

Comment:
If we are going to allow all of this, then we probably should do `{:error, status, headers, body}`.